library/classes/rulesets/Amc/reports/AMC_315g_2c/Numerator.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ *
+ * AMC_315g_2c_Numerator handles the numerator population calculation for §170.315 (g)(2) Required Test 2c which has
+ * the following requirement from MIPS for Required Test 2:
+ *
+ * For at least one unique patient seen by the MIPS eligible clinician: (1) The
+ * patient (or the patient-authorized representative) is provided timely access to
+ * view online, download, and transmit his or her health information; and (2) The
+ * MIPS eligible clinician ensures the patient's health information is available for
+ * the patient (or patient-authorized representative) to access using any
+ * application of their choice that is configured to meet the technical specifications
+ * of the Application Programming Interface (API) in the MIPS eligible clinician's
+ * certified electronic health record technology (CEHRT).
+ *
+ * We currently only support the ONC Required Test 2c version of this requirement as we do not support
+ * the View, Download, Transmit (VDT) of §170.315 (e)(1) from the patient portal.  Only clinicians can VDT on behalf
+ * of a patient.
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @copyright 2022 Discover and Change, Inc.
+ * @author    Stephen Nielson <snielson@discoverandchange.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ *
+ */
+
+use OpenEMR\Common\Database\QueryUtils;
+
+class AMC_315g_2c_Numerator implements AmcFilterIF
+{
+    public function getTitle()
+    {
+        return "AMC_315g_2c Numerator";
+    }
+
+    public function isValidPatient($date_created, $prevent_portal_access, $beginDate, $endDate)
+    {
+        // TODO: @adunsulag we will come back and visit this when we do grouping
+        if (!empty($date_created)) {
+            $creationDate = strtotime($date_created);
+            $beginDate = strtotime($beginDate);
+            $endDate = strtotime($endDate);
+            // creation date for the credentials was within the valid date boundary that we wanted
+            if ($creationDate >= $beginDate && $creationDate <= $endDate) {
+                return true;
+            }
+        }
+        // we don't worry about casing here
+        if (is_string($prevent_portal_access) && strtolower($prevent_portal_access) != "yes") {
+            // patient opted out of 3rd party portal access which makes them then eligible for 2c criteria
```

Comment:
`$prevent_portal_access != "yes"`, doesn't that mean that patient didn't opt out of 3rd party portal?

Reply:
Yes.  Good catch.  I'll fix the logic.